### DISCUSSION TITLE
What about bin width causes models not to fit/produce results?

### DISCUSSION THREAD

**Discussion by nkruskamp** [GeoStat-Framework/GSTools#144] [Q&A] [answered]
Hi again,

I'm back with a new question regarding bin width. I understand the theory behind selecting a bin width, but I'm having trouble in practice getting results to compare. My routine is to test a selection of variogram models (exponential, matern, gaussian, spherical) over a selection of bin widths. My data are relatively noisy so I'm trying to find a balance between a bin width and noise. However, for some bin widths and for some models, no models fit correctly and produce a straight lines. And as I mentioned in my previous post, I have not gotten a spherical model to produce any fit for any data yet. See attached image for an example. All models are plotted for each bin width, though they may be stacked on top of each other.

![variogram_bin-width](https://user-images.githubusercontent.com/26750564/111881459-ce052a00-8986-11eb-97d2-00befa0a6e33.png)

Could this be related to relatively uncorrelated and/or noisy data? If _not_ data related, do you have advice or guidance on how I can address this issue?

Thanks!

**Comment by MuellerSeb** [comment]
Hey there,

this looks like a fitting problem again like in #143 where the curve_fit could not find an optimal result for len_scale. Since your bins seem to be in an order of magnitude of 1000, you could try setting the inital guess for len_scale to 1000 or use the `rescale` that will come with the next release. Just have a look at the issue #143, where both approaches are described.

Hope this helps for now.
Sebastian

@LSchueler we should think about better default values for fitting.

**Comment by MuellerSeb** [comment, marked as answer]
@nfkruska Thanks to your feedback and issue #143, I came up with a new set of initial guesses for variogram fitting: #145.
`len_scale` will be set to the mean of given bin centers to obtain the correct order of magnitude. In addition, `var` and `nugget` will be set to the mean of given variogram values for the same reason.

Will be part of GSTools v1.3 :wink:

Love this! Thanks.
Sebastian

**Reply by nkruskamp** [reply]
Thanks so much! Yes that appears to be the same issue I am having. I am working in meters over a large geographic region. Scaling the units to km immediately allowed me to fit all the models as I expected including spherical.

I have also tried pixel lag scale (which is actually more appropriate for me because I need to use the models to predict in pykrige), and the scale was small enough (max lag ~300px) that the models fit appropriately.

Best!
Nick
